### DISCUSSION TITLE
labeled plucked expression

### DISCUSSION THREAD

**Discussion by yukulele** [peggyjs/peggy#406] [Q&A] [answered]
I'm not sure to understand the difference between *plucked expression* and *labeled plucked expression*.

example:
```javascript
peggy.generate('Test = @$[a]+ " "* "/" " "* @$[z]+').parse('aaa / zzz') 
// => ['aaa', 'zzz']
```
```javascript
peggy.generate('Test = @start:$[a]+ " "* "/" " "* @end:$[z]+').parse('aaa / zzz') 
// => ['aaa', 'zzz']
```
I expected to get something like `{ start: 'aaa', end: 'zzz' }` but I probably misunderstood the purpose of the labels.

**Comment by hildjj** [comment, marked as answer]
The feature you're expecting is in #302.  Right now, labeled plucked expressions are only useful for semantic predicates, as far as I know:

```peg.js
Test = @start:$[a]+ " "* "/" " "* @end:$[z]+' &{ return start.length < 1000 && end.length < 1000 }
```

**Comment by hildjj** [comment]
Going to close this as a dup of #302.

**Comment by hildjj** [comment]
#398 also adds a little more to the docs about labeled plucks.
